lib/publiccloud/instance.pm

```diff
@@ -401,20 +400,21 @@ sub wait_for_ssh {
     my $instance_msg = "instance: $self->{instance_id}, public IP: $self->{public_ip}";
     my ($duration, $exit_code, $sshout, $sysout);
 
+    # skip SLES4SAP as incompatible with get_public_ip
+    if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
+        my $public_ip_from_provider = $self->provider->get_public_ip();
+        if ($self->public_ip ne $public_ip_from_provider) {
+            record_info('IP CHANGED', "The address we know is $self->public_ip but provider returns $public_ip_from_provider");
+            $self->public_ip($public_ip_from_provider);
+        }
+    }
+
     # Looping until SSH port 22 is reachable or timeout.
     while (($duration = time() - $start_time) < $args{timeout}) {
-        $exit_code = script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1);
+        $exit_code = script_run('nc -vz -w 1 ' . $self->public_ip . ' 22', quiet => 1);
         last if (isok($exit_code) and not $args{wait_stop});    # ssh port open ok
         last if (not isok($exit_code) and $args{wait_stop});    # ssh port closed ok
 
-        # skip SLES4SAP as incompatible with get_public_ip
-        if (!get_var('PUBLIC_CLOUD_SLES4SAP')) {
-            my $public_ip_from_provider = $self->provider->get_public_ip();
-            if ($args{public_ip} ne $public_ip_from_provider) {
-                record_info('IP CHANGED', "The address we know is $args{public_ip} but provider returns $public_ip_from_provider", result => 'fail');
```

Comment:
Here the result=>fail was added to highlight and point out the change when occurred, helping to popup that event. If you don't liked it, I'd have at least changed it to `softfail`, instead of totally removing it.